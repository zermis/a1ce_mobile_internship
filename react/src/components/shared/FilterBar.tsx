import React from 'react';
import SwitchSelector, {
    ISwitchSelectorOption
} from 'react-native-switch-selector';
import Colors from '../../constants/Colors';

interface Props {
    filterMenu: ISwitchSelectorOption[];
    onPress: (value: string | number | ISwitchSelectorOption) => void;
    borderColor?: string;
    buttonColor?: string;
    textColor?: string;
    style?: any;
}

const FilterBar = (props: Props): JSX.Element => {
    return (
        <SwitchSelector
            options={props.filterMenu}
            initial={0}
            onPress={(value: string | number) => {
                props.onPress(value);
            }}
            borderColor={props.borderColor}
            buttonColor={props.buttonColor}
            textColor={props.textColor}
            fontSize={14}
            height={30}
            hasPadding
            style={props.style}
            animationDuration={110}
        />
    );
};

FilterBar.defaultProps = {
    borderColor: Colors.filterSelectorColor,
    buttonColor: Colors.filterSelectorColor,
    textColor: Colors.filterSelectorColor,
    style: { width: '100%' }
};

export default FilterBar;
