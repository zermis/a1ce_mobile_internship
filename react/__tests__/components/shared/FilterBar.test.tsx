import FilterBar from '../../../src/components/shared/FilterBar';
import renderer from 'react-test-renderer';

it('render filter bar correctly', () => {
    const tree = renderer
        .create(<FilterBar filterMenu={[]} onPress={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
