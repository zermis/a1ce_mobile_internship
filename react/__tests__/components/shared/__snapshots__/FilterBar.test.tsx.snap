// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render filter bar correctly 1`] = `
<View
  accessibilityLabel={null}
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "width": "100%",
      },
    ]
  }
  testID={null}
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 50,
          "height": 30,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#B52B37",
            "borderRadius": 50,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
          }
        }
      />
    </View>
  </View>
</View>
`;
